src/interfaces/records/messages/records-query.ts

```diff
@@ -77,8 +72,8 @@ export class RecordsQuery extends Message {
       switch (key) {
       case 'dateCreated':
         var rangeFilter: RangeFilter = {
-          gte : filter.dateCreated.from,
-          lt  : filter.dateCreated.to
+          gte : filter.dateCreated?.from!,
+          lt  : filter.dateCreated?.to!
```

Comment:
Since we're in the `case 'dateCreated'`, we should be good to `filter.dateCreated!`